pageserve/test_routes.py

```diff
@@ -66,5 +83,91 @@ def test_show_events(self):
         self.assertContext('events', EXAMPLE_EVENTS)
 
 
+class TestAddPostRoute(unittest.TestCase):
+    """Tests adding posts at POST /v1/add_post."""
+
+    def setUp(self):
+        app.app.config["TESTING"] = True
+        self.client = app.app.test_client()
+        self.expected_url = app.app.config['POSTS_ENDPOINT'] + 'add'
+
+    @patch('app.get_user', MagicMock(return_value=EXAMPLE_USER))
+    @patch('app.requests')
```

Comment:
You should use a requests mocking library like [requests-mock](https://requests-mock.readthedocs.io/en/latest/mocker.html). It turns the 4 lines for mocking a request into 1 or 2. For example, I used it here starting at line 57: https://github.com/knative-portability/large-events/blob/36f8a3e11fdf614128ed7790d8a7ea3dcc51fed5/pageserve/test_routes.py

Reply:
Done in 3e1131c